extensions/cornerstone/src/services/CornerstoneCacheService/CornerstoneCacheService.ts

```diff
@@ -67,34 +68,42 @@ class CornerstoneCacheService {
   }
 
   public async invalidateViewportData(
-    viewportData: VolumeViewportData,
+    viewportData: StackData | VolumeViewportData,
     invalidatedDisplaySetInstanceUID: string,
     dataSource,
     DisplaySetService
   ) {
     if (viewportData.viewportType === Enums.ViewportType.STACK) {
-      throw new Error('Invalidation of StackViewport is not supported yet');
-    }
-
-    // Todo: grab the volume and get the id from the viewport itself
-    const volumeId = `${VOLUME_LOADER_SCHEME}:${invalidatedDisplaySetInstanceUID}`;
+      const displaySet = DisplaySetService.getDisplaySetByUID(invalidatedDisplaySetInstanceUID)
+      if (this.stackImageIds.has(invalidatedDisplaySetInstanceUID)) {
+        this.stackImageIds.delete(invalidatedDisplaySetInstanceUID);
+        return await this._getStackViewportData(
+          dataSource,
+          [displaySet],
```

Comment:
I was confused to read this at first, then I saw our stackImageIds is a map. Can we rename it to `stackImageIdsMap`?

Reply:
(not your fault)